src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const newObject: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyForObj = item[key] as string;
+
+    if (!newObject[keyForObj]) {
+      newObject[keyForObj] = [];
+    }
+
+    newObject[keyForObj].push(item);
+  });
```

Comment:
you can rewrite it all using the reduce, it will simplify the logic and improve the readability